website/www/site/content/en/documentation/patterns/cross-language.md

```diff
@@ -0,0 +1,167 @@
+
+    def expand(self, pcoll):
+        return (pcoll
+                    | "Input preparation" >> beam.Map(lambda input: google.protobuf.text_format
+                                                .Parse(input, tf.train.Example()))
+                    | "Get predictions" >> RunInference(
+                            model_spec_pb2.InferenceSpecType(
+                                saved_model_spec=model_spec_pb2.SavedModelSpec(
+                                    model_path=model_path,
+                                    signature_name=['serving_default']))))
+
+    def to_runner_api_parameter(self, unused_context):
+        return URN, None
+
+    def from_runner_api_parameter(
+        unused_ptransform, unused_paramter, unused_context):
+        return PythonTransform()
+{{< /highlight >}}
+
+Check the [documentation](https://beam.apache.org/documentation/programming-guide/#create-x-lang-transforms) for a deeper understanding of creating an external transform.
+
+### Expansion service
+
+The expansion service is written in the same language as the external transform. It takes care of injecting the transforms in your pipeline before submitting them to the Runner.
+
+{{< highlight java >}}
+Implemented in Python.
+{{< /highlight >}}
+
+{{< highlight py >}}
+def main(unused_argv):
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+      '-p', '--port', type=int, help='port on which to serve the job api')
+  options = parser.parse_args()
+  global server
+  server = grpc.server(thread_pool_executor.shared_unbounded_instance())
+  beam_expansion_api_pb2_grpc.add_ExpansionServiceServicer_to_server(
+      expansion_service.ExpansionServiceServicer(
+          PipelineOptions(
+              ["--experiments", "beam_fn_api", "--sdk_location", "container"])),
+      server)
+  server.add_insecure_port('localhost:{}'.format(options.port))
+  server.start()
+  _LOGGER.info('Listening for expansion requests at %d', options.port)
+
+  signal.signal(signal.SIGTERM, cleanup)
+  signal.signal(signal.SIGINT, cleanup)
+  signal.pause()
+
+
+if __name__ == '__main__':
+  logging.getLogger().setLevel(logging.INFO)
+  main(sys.argv)
+{{< /highlight >}}
+
+## How to run the cross-language pipeline?
+
+In this section, the steps to run a cross-language pipeline are set out:
+
+1. Start the **expansion service** with your Python transforms: `python expansion_service.py -p 9097`
+1. Start the **Job Server** which will translated into the stage that will run on your back-end or runner (e.g. Spark):
+
+   - From Apache Beam source code:
+     `./gradlew :runners:spark:job-server:runShadow`
+   - Using the pre-build Docker container:
+     `docker run -net=host apache/beam_spark_job_server`
+
+1. **Run pipeline**: `mvn exec:java -Dexec.mainClass=CrossLanguagePipeline -Pportable-runner -Dexec.args="--runner=PortableRunner --jobEndpoint=localhost:8099 --useExternal=true --expansionServiceURL=localhost:9097 --experiments=beam_fn_api"`
```

Comment:
I'd suggest to put every option on the new line to make it more readable, something like:
```
mvn exec:java -Dexec.mainClass=CrossLanguagePipeline \
  -Pportable-runner \
  -Dexec.args=" \
    --runner=PortableRunner \
    --jobEndpoint=localhost:8099 \
    --useExternal=true \
    --expansionServiceURL=localhost:9097 \
    --experiments=beam_fn_api"
```